google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentSnapshot.java

```diff
@@ -189,51 +194,54 @@ public boolean exists() {
   }
 
   /**
-   * Returns the fields of the document as a Map. Field values will be converted to their native
-   * Java representation.
+   * Returns the fields of the document as a Map or null if the document doesn't exist. Field values
+   * will be converted to their native Java representation.
    *
-   * @return The fields of the document as a Map.
-   * @throws IllegalStateException if the document doesn't exist.
+   * @return The fields of the document as a Map or null if the document doesn't exist.
    */
-  @Nonnull
+  @Nullable
   public Map<String, Object> getData() {
-    Preconditions.checkState(exists(), "Cannot call getData() on missing document.");
-    Map<String, Object> decodedFields = new HashMap<>();
-    for (Map.Entry<String, Value> entry : fields.entrySet()) {
-      decodedFields.put(entry.getKey(), decodeValue(entry.getValue()));
+    if (fields != null) {
```

Comment:
I prefer `if (fields == null) return null` to reduce nesting, but your choice.

Reply:
Works for me.